### ISSUE TITLE
Failed to find a DNS-record for the resource at "localhost" after upgrading Node version to 18

### ISSUE THREAD

**Issue by vlad-dargel** [DevExpress/testcafe#7753]
### What is your Scenario?

I have an app that was running on Nodejs v14.17.1. Then I updated it to Node 18.14.2 and now TestCafe (v2.6.2) fails to load the pages. The only thing I can see on the network requests is that the document request returns a 302 that redirects to the about page

I have tried downgrading to Node v17 and even 16, also downgrading the TestCafe version didn't work.

Also I'm not considering it as it's experimental. I have tried setting quarentine mode, the --hostName, nothing of this worked. Also obvious to say, if I open the URL it loads perfectly on the browser outside TestCafe

### What is the Current behavior?

```
Failed to load the page at "http://localhost:8080/".
      Increase the value of the "pageRequestTimeout" variable, enable the "retryTestPages" option, or use quarantine mode to perform additional attempts to execute      
      this test.
      You can find troubleshooting information for this issue at "https://go.devexpress.com/TestCafe_FAQ_ARequestHasFailed.aspx".

Failed to find a DNS-record for the resource at "http://localhost:8080/".
```

![image](https://github.com/DevExpress/testcafe/assets/90257586/8839ba2f-9aa6-4d72-9d73-4cbc7807dda9)

### What is the Expected behavior?

-

### What is your public website URL? (or attach your complete example)

-

### What is your TestCafe test code?

--

### Steps to Reproduce

1.
2.
3.

### TestCafe version

2.6.2

### Node.js version

18.14.2

### Command-line arguments

chrome:headless --cache --allow-insecure-localhost --ignore-certificate-errors

### Browser name(s) and version(s)

Chrome

### Platform(s) and version(s)

Windows/Linux

**Comment by vlad-dargel** [comment]
> --dns-result-order=ipv4first

thanks, but it doesn't work, I get the same error `Failed to find a DNS-record for the resource at "http://localhost:8080/".`

**Comment by miherlosev** [comment]
Hi @vlad-dargel,

It's not a TestCafe issue. It's a breaking change in Node.js, which breaks the DNS address resolution mechanism.
The simplest way to fix it is changing the host name in a test page's URLs from `localhost` to `127.0.0.1`.

**Comment by vlad-dargel** [comment]
> Hi @vlad-dargel,
> 
> It's not a TestCafe issue. It's a breaking change in Node.js, which breaks the DNS address resolution mechanism. The simplest way to fix it is changing the host name in a test page's URLs from `localhost` to `127.0.0.1`.

Yes, I tried to use '127.0.0.1' instead of 'localhost' but it doesn't help too

**Comment by MattMcFarland** [comment]
@miherlosev  My team has ran into the same issue after we updated our project from node 14 to node 18.  What we found is that testcafe works fine when external domains, and it also works fine in our build pipelines.  

Want to note that our setup is not anything special, its just a gatsby js site we run on localhost during local development.

When running testcafe against localhost in node v18, we get the same error.  we also get the same error when using 127.0.0.1 - which I thought was funny because why do a DNS lookup when its an IP?   

When using `--experimental-proxyless` the problem goes away and the tests execute.   I'm not sure if this is a stable solution but it did unblock us for the time being.

**Comment by aleks-pro** [comment]
Hello @MattMcFarland ,

> The good news is when using --experimental-proxyless the problem goes away and the tests execute against localhost! :) I'm not sure if this is a stable solution but it did unblock us for the time being.

Starting with testcafe [v3.0.0](https://testcafe.io/404438/release-notes/framework/2023-6-21-introducing-testcafe-v-3-0-0#native-automation), the Native Automation mode aka Proxyless is enabled by default. Currently, it has several [limitations](https://testcafe.io/documentation/404237/guides/intermediate-guides/native-automation-mode#q-what-are-the-limitations-of-native-automation), but if they do not affect your tests, you can use this mode to overcome the issue.

**Comment by yasinkocak** [comment]
Any news of this issue, i get the same error

**Comment by PavelMor25** [comment]
Hello @yasinkocak,
 
Native automation must fix the issue. If you can reproduce the issue using native automation mode, please create a separate ticket using this [template.](
https://github.com/DevExpress/testcafe/issues/new?assignees=&labels=TYPE%3A+bug&template=bug_report.yaml)
